src/SFML/Graphics/Shader.cpp

```diff
@@ -40,6 +40,18 @@
 
 #ifndef SFML_OPENGL_ES
 
+#if defined(SFML_SYSTEM_MACOS) || defined(SFML_SYSTEM_IOS)
+
+    #define castToGlHandle(x) reinterpret_cast<GLEXT_GLhandle>(static_cast<ptrdiff_t>(x))
+    #define castFromGlHandle(x) static_cast<unsigned int>(reinterpret_cast<ptrdiff_t>(x))
+
+#else
+
+    #define castToGlHandle(x) (x)
+    #define castFromGlHandle(x) (x)
+
+#endif
```

Comment:
I vote for the macros. We're wasting so much time overengineering this, just for the beauty of C++... Let's be pragmatic and try to make more useful comments on this huge PR :stuck_out_tongue: